CONTRIBUTING.md

```diff
@@ -0,0 +1,223 @@
+# Contributing
+
+This repo is built and maintained by the Galoy team. We welcome and appreciate new contributions and encourage you to check out the repo and [join our community Slack](https://join.slack.com/t/galoymoney-workspace/shared_invite/zt-rvnhsdb5-72AZCD_jzw6_Q05aCs0SgA) to get started.
+
+To help get you started, we will explain a bit about how we've laid out the code here and some of the practices we use. Our layout details fall mostly into two categories:
+- [Working with Types](#working-with-types)
+- [Understanding our Architecture](#architecture)
+
+---
+
+## Working with Types
+This codebase is implemented using a well-typed approach where we rely heavily on the Typescript compiler for catching & enforcing certain kinds of checks we would like to perform.
+
+This approach helps to strenghthen guarantees that bugs/issues don't get to runtime unhandled if our types are implemented properly.
+
+### Our Approach
+We define all our types in declaration files (`*.d.ts`) where typescript automatically adds any types to the context without needing to explicitly import. Declaration files have limitations on what can be imported & instantiated within them, so in the few cases where types derive from instances of things we would usually have to use special tricks to derive them.
```

Comment:
why are we using this approach?

reading on https://stackoverflow.com/questions/21247278/about-d-ts-in-typescript, it seems the approach of using `.d.ts` is for library that are not typescript native

Reply:
@samerbuna advocated for this. The reason was so that we don't have to explicitly import types everywhere I believe. It works well from a dev flow approach so far in my experience.